src/providers/gdal/qgsgdaldataitems.cpp

```diff
@@ -99,7 +99,9 @@ QVector<QgsDataItem *> QgsGdalLayerItem::createChildren()
 
       childItem = new QgsGdalLayerItem( this, name, mSublayers[i], mSublayers[i] );
       if ( childItem )
-        this->addChildItem( childItem );
+      {
```

Comment:
@wonder-sk would you please have a look to this commit? I'm not sure why it was calling directly `addChildItem` instead of adding the item to the list of `children`